### PR TITLE
feat: add first poc for a Kafka proxy

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR adds contains the first POC of a Kafka Proxy that validates messages. As a poc, there has no real validation logic behind but the functionality is ready to make it happen: Right now it only prints a log line whenever a message is **literally** the string `invalid message`.

The PR also includes a `docker-compose.yml` and a `Dockerfile` so we can run a one node Kafka cluster and do some testing. Readme update is pending.

If you wanna play in the meantime, you can just run `docker compose up` and produce/consume messages by:

Consume: `docker compose exec broker /bin/kafka-console-consumer --topic demo --from-beginning --bootstrap-server proxy:28002`
Produce: `docker compose exec broker /bin/kafka-console-producer --topic demo --bootstrap-server proxy:28002` and start writing messages (stdin).

As notes for developers, I managed to completely turn off my old POC (yeah, never pushed but I can share it with you as I have it locally) where I had to manage almost all the protocol handling manually (yeah, dealing with bits and bytes) and which was only supporting a particular Kafka request protocol version. 
Now I'm using https://github.com/grepplabs/kafka-proxy which it has almost everything we need. In fact I had to fork it and tune it up a bit (Adding support for request handlers per Key. See the fork [here](https://github.com/smoya/kafka-proxy/blob/request-key-handlers/proxy/processor.go#L203-L231). I plan to create a PR to the main project once I have more time to improve those changes I did.
I'm also using parts of the code of https://github.com/Shopify/sarama project for decoding and validating the `ProduceRequest`. I could not use the project directly because some of the interfaces I needed are private. I opened an issue few days ago regarding this: https://github.com/Shopify/sarama/issues/1967.

One of the biggest achievements on using https://github.com/grepplabs/kafka-proxy  is that it supports most of the AUTH mechanisms and also all the Kafka protocol request versions (YEEEEI!).

**Next steps**
- The plan is, from here, to start validating actual AsyncAPI spec files somehow

**How to review this PR**
Even though this is a POC, most of this code could be used as the foundation of the first versions of the gateway.
That means you should review it as any other PR. Some things that were not included on this PR with the idea of doing it in another one are:

- Tests of the logic in `main.go`
- README update

Some things that were explicitly excluded from this PR were:

- Tests for code copied from Sarama (most of the code in `/kafka`)

**Related issue(s)**
https://github.com/asyncapi/shape-up-process/issues/70

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+linters:
```
Comment (smoya) [inline .golangci.yml:1]: Comes from https://github.com/asyncapi/template-for-go-projects

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Makefile based on https://gist.github.com/thomaspoignant/5b72d579bd5f311904d973652180c705
```
Comment (smoya) [inline Makefile:1]: Comes from https://github.com/asyncapi/template-for-go-projects

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	github.com/xdg/scram v1.0.3 // indirect
+	github.com/xdg/stringprep v1.0.3 // indirect
+	golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b // indirect
+	golang.org/x/net v0.0.0-20210427231257-85d9c07bbe3a // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+)
+
+replace github.com/grepplabs/kafka-proxy => github.com/smoya/kafka-proxy v0.2.9-0.20210623125118-a94cf71a065c
```
Comment (smoya) [inline go.mod:25]: Here is the replace directive in order to use my forked version of grepplabs/kafka-proxy

File: .gitignore [hunk 4]
```diff
@@ -8,8 +8,11 @@
 # Test binary, built with `go test -c`
```
Comment (smoya) [inline .gitignore:8]: Comes from https://github.com/asyncapi/template-for-go-projects

File: main.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
+package main
```
Comment (smoya) [inline main.go:1]: This file is actually the code of the app :)

**Review by fmvilas — APPROVED** [review]
:shipit:

**Pull request merged**